### PR TITLE
Adds CRUD for characters

### PR DESCRIPTION
**Pull request by AdamM-FC**
This PR adds the create, read, update and delete for characters. Also adds unit tests for each request

### CODE REVIEW

File: app/controllers/characters_controller.rb [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+    json_response(@character, :created)
   end
 
   def show
+    json_response(@character)
   end
 
   def update
+    @character.update(character_params)
+    head :no_content
```
Comment (MassimoPolimeni) [inline app/controllers/characters_controller.rb:20]: 🤔 we don't want to return the JSON updated?

File: spec/requests/characters_spec.rb [hunk 2]
```diff
@@ -1,15 +1,101 @@
...
+  # PUT
+  describe 'Put /characters/:id' do
+    let(:valid_attributes) { { name: 'test' } }
+
+    context 'when the record exists' do
+      before { put "/characters/#{character_id}", params: valid_attributes }
+
+      it 'updates the record' do
+        expect(response.body).to be_empty
+      end
```
Comment (MassimoPolimeni) [inline spec/requests/characters_spec.rb:85]: maybe it's worthing check if the `name` was updated? but this is related to my previous comment, if we only return **200** with no JSON there's no quick way to check this 🙅‍♂️

**Review by MassimoPolimeni — APPROVED** [review]
Amazing first steps with RoR 👏 
**LGTM**, left just a though around the UPDATE 🤷‍♂️ not a blocker!

File: app/controllers/characters_controller.rb (hunk 1 above)
Comment (AdamM-FC) [reply app/controllers/characters_controller.rb:20]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204#:~:text=The%20common%20use%20case%20is,200%20should%20be%20used%20instead.

File: spec/requests/characters_spec.rb (hunk 2 above)
Comment (AdamM-FC) [reply spec/requests/characters_spec.rb]: Updated

**Review by mugx-fc — APPROVED** [review]
Looks good 👍

File: spec/requests/characters_spec.rb [hunk 3]
```diff
@@ -1,15 +1,108 @@
...
+      before { put "/characters/#{character_id}", params: valid_attributes }
+
+      it 'returns empty body' do
+        expect(response.body).to be_empty
+      end
+
+      it 'updates the record' do
+        character = Character.find(character_id)
+        expect(character.name).to eq(name)
+      end
```
Comment (MassimoPolimeni) [inline spec/requests/characters_spec.rb:92]: 👏

**Pull request merged**